meter/tesla.go

```diff
@@ -70,21 +88,32 @@ func NewTeslaFromConfig(other map[string]interface{}) (api.Meter, error) {
 		cc.Usage = "solar"
 	}
 
-	return NewTesla(cc.URI, cc.Usage)
+	return NewTesla(uri, cc.Usage, cc.Password)
 }
 
 // NewTesla creates a Tesla Meter
-func NewTesla(uri, usage string) (api.Meter, error) {
+func NewTesla(uri *url.URL, usage string, password string) (api.Meter, error) {
 	log := util.NewLogger("tesla")
 
+	uri.Scheme = "https"
```

Comment:
if someone has a proxy in front (for whatever reason) that will kill the access. I'd prefer to leave users the choice asimplemented  before.